### ISSUE TITLE
Create tool to vacate drive

### ISSUE THREAD

**Issue by jayroos** [trapexit/mergerfs-tools#55]
It would be helpful to have a tool that could be used to vacate a drive in a pool to prepare it for removal or replacement. Assuming the pool has sufficient capacity to hold the files from the target drive, the utility would redistribute the files in a similar manner to mergerfs.balance.

If there is already in-built support for this, I'm happy to hear about it.

Thanks!

**Comment by trapexit** [comment]
I suppose a tool could be created. I've always just created a second pool which excludes the drive in question then rsync into that pool. Allows for picking the specific policy which my be necessary since you probably want to keep the existing policy while you continue to use the pool but want data spread differently when writing. That becomes far more complicated with an app since it'd need to replicate all that builtin logic. mergerfs.balance ignores the pool's policies and just rsyncs files around underneath mergerfs.

**Comment by jayroos** [comment]
I don't know enough about how everything works, but here is one thought.

What if the vacate tool was a variant of balance that created large zero-fill temp files on the drive to be vacated which would be skipped during the balance phase. Then the cycle would look like this:

1. Fill drive with temp files
2. Balance
3. Return to #1 until no non-temp files remain.

One could probably do this manually, but having it all in one tool would be slick.

Maybe not the best approach, but would it work?

**Comment by trapexit** [comment]
I really don't follow. The balance tool works independent of mergerfs outside querying it for the drives. It ignores ignores the policies the pool have completely. It's a simple way to redistribute files when you don't care at all which drive they end up on.

Vacating a drive is different. At least it always is for me. I more often than not to obey the policies. Which means I need to use mergerfs to make the decisions. That can only be done by creating another mount and moving files into it from the drive in question.

If you don't care about where files end up then changing the balance tool to use the drive as the source rather than the drive with the most used space is easy. But in my opinion so is removing the drive from the pool and running rsync.

**Comment by jayroos** [comment]
I'm trying to find a way to do it while "online". Removing the drive from the pool effectively takes that data "offline" until the rsync is done.

**Comment by trapexit** [comment]
That's why I said use a *second* pool.

**Comment by jayroos** [comment]
Let me see if I understand:

1. Create pool2 excluding the drive to be removed. Both pools are active simultaneously.
2. Rsync from pool1 to pool2
3. Remove pool2
4. Remove drive to be removed from pool1

**Comment by trapexit** [comment]
No. Not rsync from pool1 to pool2. Rsync from excluded drive to pool2.

**Comment by trapexit** [comment]
This is generally documented in my extra docs.

https://github.com/trapexit/backup-and-recovery-howtos/blob/master/docs/recovery_(mergerfs).md

**Comment by jayroos** [comment]
Thanks. I missed that because I didn't think of it as recovery.

**Comment by eduncan911** [comment]
I would like to vote for this as well.

I've read the link above and understand how to add another drive, and move data over.

I suspect to just "vacate" a drive, I would:

* first remove it from the pool
* second, copy/rsync the data from the drive i just removed back into the pool (to continue the use of its policy settings)
* third, once copy is completed, then remove the physical drive

I think the OP's original thought would be for mergefs itself to handle this, if you remove the drive.  Like an option or flag to move all the data.

However, I also see the author's point: that mergefs is just a proxy service or sort.  And there are these tools here in this repo for more out-of-band functionality.  one could write a simple tool (and PR it here) for exactly that.  maybe `mergerfs.remove` or `mergerfs.vacate` or something.

I think it would come down to this:

If `mergefs` is intended to be responsible for maintenance of the pool at all, then yes this feature should be added.

But right now, reading the `mergefs` docs and reviewing these Tools, it would seem to be that:

* `mergefs` is intended to be a simple proxy, perhaps to keep the code simple (it's got a lot of files already!).  
* And maintenance of the pool remains dedicated to these tools here in this repo.

Disclaimer: I just found out about `mergefs` and think it finally ticks most of the boxes I've been looking for to move off of DrivePool and back to full Linux servers again.  Especially for Plex and media.  I'm sitting on 15 or so disks and nearly 60 TB of space, largely duplicated 2x to 3x over.  I'm still reading on it but I think I am ready to take the plunge!

**Comment by celogeek** [comment]
I'm doing it right now. 
I have 3 disk of 1TB and one of 2TB. 
I want to split my 2 TB disk into 2 partition of 1 TB.
What I do in a non efficient way is:
- remove the disk from the pool
- rsync the disk of 2 TB to the pool
- repartition the disk
- add back the 2 partitions to the pool

during the process a part of my data is missing. 
I could have done it a bit more manually by moving the data from this disk to others also. 

A tools to empty a disk from a pool to the other disks of the pool could be helpful.

**Comment by trapexit** [comment]
The biggest thing that needs to be done to make it practical is the ability to get all options and create a list of them compatible for a mounting as the only good way to manage this behavior is to create a second, temporary pool without the drive in question and then rsync. Shouldn't be hard just needs to be done.
